client/models/scenery_editor.js

```diff
@@ -318,16 +321,76 @@ class SceneryEditor {
             post('api/tools/scenery/instance/delete/' + WORKSPACES.opened, {
                 id: document.getElementById('tools-detail-scenery-id').innerText
             }, () => {
-                WORKSPACES.reload();
+                this.removeObject(document.getElementById('tools-detail-scenery-id').innerText);
             });
         } else if (this.selected_type === 'unique') {
             post('api/tools/scenery/unique/delete/' + WORKSPACES.opened, {
                 id: document.getElementById('tools-detail-scenery-id').innerText
             }, () => {
-                WORKSPACES.reload();
+                this.removeUnique(document.getElementById('tools-detail-scenery-id').innerText);
             });
         }
     }
+
+    addObject(object){
+        let object_id = object.x + ',' + object.y;
+            
+        WORKSPACES.current_map.sceneryLoader.createScenery(object, (model, definition) => {
+            let m = createSceneryMesh(object_id, object, WORKSPACES.current_map.terrain, model, definition);
+            m.original_id = { type: 'scenery', id: object_id };
+            WORKSPACES.current_map.scenery_groups['trees'].add(m);
+            WORKSPACES.current_map.scenery_references[object_id] = {
+                instance: object,
+                definition: definition,
+                threeObject: m
+            }
+        })
+    }
+
+    editObjectTint(object){
+        let newObject = WORKSPACES.current_map.scenery_references[object.id].instance;
+
+        this.removeObject(object.id)
+        this.applyObjectMod(newObject, object.modification, this._tintValueCheck(object))
+        this.addObject(newObject)
+    }
+
+    _tintValueCheck(object){
+        return (object.tint) ? object.tint : object.remove_tint;
+    }
+
+    editObjectRotation(object){
+        let newObject = WORKSPACES.current_map.scenery_references[object.id].instance;
+
+        this.removeObject(object.id)
+        this.applyObjectMod(newObject, object.modification, object.rotation)
+        this.addObject(newObject)
+    }
+
+    applyObjectMod(object, modification, value){
```

Comment:
This method is completely unnecessary and an example of where 'factoring out code' goes too far.